joint_state_controller:
  type: "joint_state_controller/JointStateController"
  publish_rate: 300
arm:
  arm_joint_controller:
    type: "effort_controllers/JointTrajectoryController"
    joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
    gains:
      joint1: { p: 40.0, i: 1.0, d: 0.15, i_clamp: 4.0 }
      joint2: { p: 460.0, i: 57.0, d: 7.0, i_clamp: 100.0 }
      joint3: { p: 290.0, i: 28.0, d: 5.0, i_clamp: 50.0 }
      joint4: { p: 22.0, i: 0.05, d: 0.15, i_clamp: 4.0 }
      joint5: { p: 65.0, i: 1.5, d: 0.1, i_clamp: 4.0 }
      joint6: { p: 6.0, i: 0.05, d: 0.1, i_clamp: 4.0 }

  arm_gripper_controller:
    type: "effort_controllers/JointTrajectoryController"
    joints:
      - joint7
    gains:
      joint7:
        {
          p: 15000,
          i: 1500,
          d: 0.0,
          i_clamp_max: 600.0,
          i_clamp_min: -600.0,
          antiwindup: true,
        }
